### PR TITLE
Lt 5595 migrate to net 8 new

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/MarginTrading.Backend.Services/AssetPairs/ScheduleSettingsCacheService.cs [hunk 1]
```diff
@@ -231,7 +231,7 @@ public Dictionary<string, List<CompiledScheduleTimeInterval>> GetMarketsTradingS
 
             try
             {
-                return _compiledMarketScheduleCache.ToDictionary();
+                return MoreEnumerable.ToDictionary(_compiledMarketScheduleCache);
```
Comment (gponomarev-lykke) [inline src/MarginTrading.Backend.Services/AssetPairs/ScheduleSettingsCacheService.cs:234]: In the [AssetPairsCache] class MoreEnumerable is replaced by 
```
return pairsEnum.DistinctBy(p => p.Item1).ToDictionary();
```

Here it's the opposite, the normal enumerable replaced by MoreEnumerable
Why is this needed?

File: src/MarginTrading.Backend.Services/AssetPairs/ScheduleSettingsCacheService.cs [hunk 2]
```diff
@@ -358,7 +358,7 @@ public Dictionary<string, MarketState> GetMarketState()
 
             try
             {
-                return _marketStates.ToDictionary();
+                return MoreEnumerable.ToDictionary(_marketStates);
```
Comment (gponomarev-lykke) [inline src/MarginTrading.Backend.Services/AssetPairs/ScheduleSettingsCacheService.cs:361]: Same here

File: tests/MarginTradingTests/MarginTradingTests.csproj [hunk 3]
```diff
@@ -1,54 +1,55 @@
...
+<Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+        <AssemblyName>MarginTradingTests</AssemblyName>
+        <PackageId>MarginTradingTests</PackageId>
+        <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
+        <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
+        <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
+        <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
+        <Version>2.32.0</Version>
```
Comment (gponomarev-lykke) [inline tests/MarginTradingTests/MarginTradingTests.csproj:10]: Previous version was 
<Version>2.34.0</Version>

File: src/MarginTrading.Backend.Services/AssetPairs/ScheduleSettingsCacheService.cs (hunk 1 above)
Comment (tarurar) [reply src/MarginTrading.Backend.Services/AssetPairs/ScheduleSettingsCacheService.cs:234]: It will not be replaced. The MoreLinq extension was used here before so as not to change the behaviour, I continued to use it. A semantic change is required due to a conflict between MoreLinq's implementation of the `ToDictionary` method and the new implementation of `System.Linq` in NET 8.

**Review by gponomarev-lykke — APPROVED** [review]
